src/v/container/include/container/fragmented_vector.h

```diff
@@ -288,10 +378,28 @@ class fragmented_vector {
 
         iter(vec_type* vec, size_t index)
           : _index(index)
-          , _vec(vec) {}
+          , _vec(vec) {
+#ifndef NDEBUG
+            // NOLINTNEXTLINE(cppcoreguidelines-prefer-member-initializer)
+            _my_generation = vec->_generation;
```

Comment:
nit: why not as member initializer?

Reply:
Because I find this easier, but I can reorder the member variables so it's less awkward to conditionally compile.

Reply:
Fixed.